### PR TITLE
Change formatting of colorscheme doc to markdown

### PR DESCRIPTION
**Pull request by parmort**
Makes the file easier to read in github while still maintaining the
current easy-to-read format found in the raw file

#### ISSUE TYPE

- Improvement/feature implementation

#### CHECKLIST

- [X] The `CONTRIBUTING` document has been read **[REQUIRED]**
- [X] All changes follow the code style **[REQUIRED]**
- [X] All new and existing tests pass **[REQUIRED]**
- [ ] Changes require config files to be updated
    - [ ] Config files have been updated
- [X] Changes require documentation to be updated
    - [X] Documentation has been updated
- [ ] Changes require tests to be updated
    - [ ] Tests have been updated

#### DESCRIPTION

The original `colorschemes.txt` has been deleted in favor of a
github-flavored markdown file.

#### MOTIVATION AND CONTEXT

This makes the `colorschemes` documentation easier to read in github,
where most of the traffic to that file will occur.

### CODE REVIEW

File: doc/colorschemes.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Colorschemes
+
+This text explains colorschemes and how they work.
+
+## Context Tags
+
+Context Tags provide information about the context. If the tag `in_titlebar` is
+set, you probably want to know about the color of a part of the titlebar now.
+
+There are a number of context tags, specified in /ranger/gui/context.py in the
```
Comment (toonn) [inline doc/colorschemes.md]: You missed at least one path.

File: doc/colorschemes.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Colorschemes
```
Comment (toonn) [inline doc/colorschemes.md]: Could you switch back to the underline style headings, they're valid markdown and superior in plaintext form imo.

File: doc/colorschemes.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+## Context Tags
+
+Context Tags provide information about the context. If the tag `in_titlebar` is
+set, you probably want to know about the color of a part of the titlebar now.
+
+There are a number of context tags, specified in /ranger/gui/context.py in the
+constant `CONTEXT_KEYS`.
+
+A Context object, defined in the same file, contains attributes with the names
+of all tags, whose values are either True or False.
```
Comment (toonn) [inline doc/colorschemes.md]: Values like `True` and `False` should be backticked.

**Comment by parmort** [comment]
Hope that helps a bit. The review bit got all jumbled because I squashed the commits. Oops.

File: doc/colorschemes.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Context Tags provide information about the context. If the tag `in_titlebar` is
+set, you probably want to know about the color of a part of the titlebar now.
+
+There are a number of context tags, specified in `/ranger/gui/context.py` in the
+constant `CONTEXT_KEYS`.
+
+A Context object, defined in the same file, contains attributes with the names
+of all tags, whose values are either `True` or `False`.
+
+Implementation in the GUI Classes
```
Comment (toonn) [inline doc/colorschemes.md]: Looks like you missed a heading. Probably got lost in the squashing.

File: doc/colorschemes.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+The class CursesShortcuts in the file `/ranger/gui/curses_shortcuts.py` defines
```
Comment (toonn) [inline doc/colorschemes.md]: Classes should be `verbatim` too.

File: doc/colorschemes.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+the methods `color(*tags)`, `color_at(y, x, wid, *tags)` and `color_reset()`.
+This class is a superclass of Displayable, so these methods are available almost
```
Comment (toonn) [inline doc/colorschemes.md]: Displayable's another class.

File: doc/colorschemes.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+Colorschemes
+============
+
+This text explains colorschemes and how they work.
+
+Context Tags
+------------
+
+Context Tags provide information about the context. If the tag `in_titlebar` is
```
Comment (toonn) [inline doc/colorschemes.md]: Let's lowercase the "Tags".

File: doc/colorschemes.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+The class CursesShortcuts in the file `/ranger/gui/curses_shortcuts.py` defines
+the methods `color(*tags)`, `color_at(y, x, wid, *tags)` and `color_reset()`.
+This class is a superclass of Displayable, so these methods are available almost
+everywhere.
+
+Something like `color("in_titlebar", "directory")` will be called to get the
+color of directories in the titlebar. This creates a `ranger.gui.context.Context`
+object, sets its attributes `in_titlebar` and `directory` to True, leaves the
+others as False, and passes it to the colorscheme's `use(context)` method.
```
Comment (toonn) [inline doc/colorschemes.md]: Missed False.

File: doc/colorschemes.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+----------------
+
+A colorscheme should be a subclass of `ranger.gui.ColorScheme` and define the
+method `use(context)`. By looking at the context, this use-method has to
+determine a 3-tuple of integers: `(foreground, background, attribute)` and return
+it.
+
+`foreground` and `background` are integers representing colors, `attribute` is
+another integer with each bit representing one attribute. These integers are
+interpreted by the used terminal emulator.
```
Comment (toonn) [inline doc/colorschemes.md]: used term em -> term em in use

File: doc/colorschemes.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+Abbreviations for colors and attributes are defined in `ranger.gui.color`. Two
+attributes can be combined via bitwise OR: `bold | reverse`
+
+Once the color for a set of tags is determined, it will be cached by default. If
+you want more dynamic colorschemes (such as a different color for very large
+files), you will need to dig into the source code, perhaps add an own tag and
```
Comment (toonn) [inline doc/colorschemes.md]: an own tag -> a custom tag

**Comment by toonn** [comment]
Thanks for your first contribution! 🎉
Keep 'em coming ; )

**Pull request merged**